docs/modules/ROOT/pages/mgmt_services/elastic_ip_manager.adoc

```diff
@@ -0,0 +1,207 @@
+= Elastic IP Manager Design Specification
+Eric Yuan <dakuaiera@sina.com>, Liguang Xie <lxie@futurewei.com>
+v0.1, 2020-05-24
+:toc: right
+:imagesdir: ../../images
+
+NOTE: This document is under development
+
+== Overview
+
+The Elastic IP (EIP) manager is responsible of allocating and releasing EIP as well as associating the allocated EIP to a running instance or network interface (port).
+This is the acting microservice that provides elastic/floating IP APIs to external customers (through API gateway) and driving the elastic IP CURD workflow.
+A network interface with an enabled EIP could establish inbound connections from and outbound connections to external networks like Internet.
+
+=== User Cases
+A customer of Alcor could leverage Elastic IP manager in the following cases:
+
+* Allocate or release a _standalone_ EIP within the user account.
+* Associate or disassociate an EIP to a running instance, which by default attach to its primary interface.
+* Associate or disassociate an EIP to a network interface.
+
+=== Service Basic Requirements
+
+[arabic]
+1. Provide CURD REST APIs for customers or provider admins to management EIPs. The APIs are compatible to OpenStack Neutron v2.0 APIs <<neutron_api>>.
+1. Provide CURD REST APIs for cloud provider admins to register or unregister a list of public IPv4 or IPv6 ranges in EIP manager.
+Note that those IP ranges should be owned by cloud provider.
+1. Allocate _standalone_ and _unique_ EIP address. Guarantee uniqueness of EIP address in a regional scope.
```

Comment:
should the uniqueness be in the world level?

Reply:
An EIP is a resource unique globally but the EIP service in one region has no visibility on other EIP services. Therefore from EIP service perspective, it only guarantee region-wide uniqueness while the global uniqueness is ensured by additional mechanisms.